flang/test/Lower/OpenMP/omp-sections.f90

```diff
@@ -0,0 +1,51 @@
+! This test checks the lowering of OpenMP sections construct with several clauses present
```

Comment:
Please add more tests to check handling of nowait, optional omp section etc.

Reply:
I added this test. As well as added the clauses we plan to support in future. Is there anything else that could be added?

Reply:
(Correction: It does check the nowait clause, did not observe it at the end)

Reply:
At the moment we do not have support for privatisation in the OpenMP dialect. We tried to make some progress with and RFC but did not reach a conclusion (https://llvm.discourse.group/t/rfc-privatisation-in-openmp-dialect/3526). So we decided to implement privatisation in the lowering itself. Currently we have initial support for `private` and `firstprivate`. I have provided a comment in the review on where to handle this. 

I think we can add TODO error messages for reduction, lastprivate. 

Note: Once we have more visibility on types with constructors/destructors(final) we can investigate further on handling privatisation in the OpenMP dialect.